subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -110,6 +117,69 @@ private static void showUsage(PrintStream out, CommandLineParser parser) {
         out.println();
     }
 
+    private static class WelcomeMessageAction implements Action<PrintStream> {
+        private final BuildLayoutParameters buildLayoutParameters;
+
+        public WelcomeMessageAction(BuildLayoutParameters buildLayoutParameters) {
+            this.buildLayoutParameters = buildLayoutParameters;
+        }
+
+        @Override
+        public void execute(PrintStream out) {
+            GradleVersion currentVersion = GradleVersion.current();
+            File gradleUserHomeDir = buildLayoutParameters.getGradleUserHomeDir();
+            File notificationsDir = new File(gradleUserHomeDir, "notifications/" + currentVersion.getVersion());
```

Comment:
I would guess that we need to normalize `"notifications/"` for windows

Reply:
The test on Windows seems to pass just fine. Either way I changed the code to build up a hierarchy of file references instead.

Reply:
Cool, I see other places also hard code `/` and it works. This must be a feature of the `File` API